spec/features/adding_items_to_cart_spec.rb

```diff
@@ -155,6 +153,117 @@
         end
       end
     end
+
+    context "when both of the bundle items have a user-selectable variant", js: true do
+      let(:bundle) { create(:product_in_stock, name: "Bundle", sku: "BUNDLE") }
+      let(:keychain) do
+        create(:product_in_stock, name: "Keychain",
+                                  sku: "KEYCHAIN",
+                                  can_be_part: true)
+      end
+
+      let(:shirt) do
+        bundled_product_from_options(
+          name: "Shirt", option_type: "Size", option_values: ["Large", "XL"]
+        )
+      end
+      let(:hat) do
+        bundled_product_from_options(
+          name: "Hat", option_type: "Color", option_values: ["Red", "Blue"]
+        )
+      end
+
+      before do
+        add_part_to_bundle(
+          bundle,
+          keychain.master,
+          count: 1
+        )
+
+        add_part_to_bundle(
+          bundle,
+          shirt.master,
+          variant_selection_deferred: true
+        )
+
+        add_part_to_bundle(
+          bundle,
+          hat.master,
+          variant_selection_deferred: true
+        )
+      end
+
+      context "and the user selects differing variants from the existing line item" do
+        specify "the cart contains incremented variants when listing bundle items" do
+          add_item_to_cart(size: "Large", color: "Red")
+          add_item_to_cart(size: "XL", color: "Blue")
+
+          within all("#cart-detail .line-item")[0] do
```

Comment:
Just checking that the line items markup does not have any other unique identifiers to be used as clearer selectors for Capybara...